appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java

```diff
@@ -203,11 +203,12 @@ private void sendProof() {
 
         Log.e(TAG, "Proof " + (proofsSent + 1) + " skipped! Came from background!");
       } else {
-        // Connection to service may already been done, but we still need to make sure that it is
-        // connected. In case no connection is not yet done, the message is stored to be sent as soon as
-        // the connection is done.
-        poaConnector.connectToService(appContext);
-        // send proof
+        if (!firstProof) {
+          Log.e(TAG, "Start proof sending");
+          postponeSendProof();
+          firstProof = true;
+          return;
```

Comment:
Don't use the return to stop the process when you are in the middle of several conditions that are not using return statement to finish it, just use if else to do it.